app/src/main/java/org/sil/gatherwords/EntryActivity.java

```diff
@@ -297,4 +310,35 @@ public PlayButton(Context ctx) {
         }
     }
 
+
+    private static class AddNewWordToDB extends AsyncTask<Void, Void, Void> {
+        WeakReference<EntryActivity> entryActivityRef;
+        WordDao wDAO;
+        Long sessionID;
+
+        AddNewWordToDB(EntryActivity entryActivity) {
+            entryActivityRef = new WeakReference<>(entryActivity);
+            wDAO = AppDatabase.get(entryActivity).wordDao();
+            sessionID = entryActivity.sessionID;
+        }
+
+        @Override
+        protected Void doInBackground(Void... v) {
+            Word newWord = new Word();
+            newWord.sessionID = sessionID;
+            wDAO.insertWord(newWord);
```

Comment:
Also need to create associated `Meaning` entries.  Each of these represents an editable field for the word.  Create one `Meaning` for each option enabled in the Preferences.  Wrap all of these inserts into a transaction.